morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -463,13 +399,30 @@ public void ensureIndexes(final boolean background) {
         return getByKeys(null, keys);
     }
 
+    private DBCollection getCollection(final Object obj) {
+        if (obj == null) {
+            return null;
```

Comment:
Since this method has to made public again anyway to preserve binary compat, best to leave the behavior as is since applications may be relying on it.